multiversx_sdk_cli/cli_wallet.py

```diff
@@ -107,8 +112,24 @@ def wallet_new(args: Any):
     format = args.format
     outfile = args.outfile
     address_hrp = args.address_hrp
+    shard = args.shard
+
+    if shard in CURRENT_SHARDS:
+        i = 0
+        while i < MAX_ITERATIONS_FOR_GENERATING_WALLET:
```

Comment:
Can also be a for from 0 to `max iterations`, with a break within (as you did now).

Reply:
changed to `for` loop